Client/Assets/Passwords.js

```diff
@@ -0,0 +1,699 @@
+/* vim: set ts=2 sts=2 sw=2 et tw=80: */
+/* map Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+"use strict";
+
+var gEnabled = true;
+var gStoreWhenAutocompleteOff = false;
+var gAutofillForms = true;
+var gDebug = false
+
+function log(pieces) {
+  function generateLogMessage(args) {
+    var strings = ['Login Manager (content):'];
+
+    args.forEach(function(arg) {
+      if (typeof arg === 'string') {
+        strings.push(arg);
+      } else if (typeof arg === 'undefined') {
+        strings.push('undefined');
+      } else if (arg === null) {
+        strings.push('null');
+      } else {
+        try {
+          strings.push(JSON.stringify(arg, null, 2));
+        } catch(err) {
+          strings.push("<<something>>");
+        }
+      }
+    });
+    return strings.join(' ');
+  }
+
+  if (!gDebug)
+    return;
+
+  var message = generateLogMessage(pieces);
+  // Turn off logging, since its visible to the page
+  // dump(message + "\n");
+  // console.log(message);
+}
+
+var LoginManagerContent = {
```

Comment:
There's actually a diff of the two files in the bug. Its substantial, but not a crazy amount. I could done a little less if we wanted, but I think Desktop's Password code is actually going to be basically thrown out and rewritten soon.